src/hw/connectApp.js

```diff
@@ -154,19 +160,28 @@ const cmd = ({
   devicePath,
   appName,
   requiresDerivation,
+  dependencies,
 }: Input): Observable<ConnectAppEvent> =>
   withDevice(devicePath)((transport) =>
     Observable.create((o) => {
       const timeoutSub = of({ type: "unresponsiveDevice" })
         .pipe(delay(1000))
         .subscribe((e) => o.next(e));
 
-      const sub = defer(() => from(getAppAndVersion(transport)))
-        .pipe(
+      const innerSub = ({ appName, dependencies }: any) =>
+        defer(() => from(getAppAndVersion(transport))).pipe(
           concatMap((appAndVersion): Observable<ConnectAppEvent> => {
             timeoutSub.unsubscribe();
 
             if (isDashboardName(appAndVersion.name)) {
+              // check if we meet dependencies
+              if (dependencies?.length) {
+                return streamAppInstall({
+                  transport,
+                  appNames: [appName, ...dependencies],
+                  onSuccessObs: () => innerSub({ appName }), // NB without deps
```

Comment:
Return the same thing we just did, but without the dependency array since we completed the inline install